airbyte-server/src/main/java/io/airbyte/server/handlers/ConnectionsHandler.java

```diff
@@ -140,6 +141,34 @@ public ConnectionRead createConnection(final ConnectionCreate connectionCreate)
       standardSync.withCatalog(new ConfiguredAirbyteCatalog().withStreams(Collections.emptyList()));
     }
 
+    if (connectionCreate.getSchedule() != null && connectionCreate.getScheduleType() != null) {
+      throw new JsonValidationException("supply old or new schedule schema but not both");
```

Comment:
Indeed we mostly handle one or the other or both (e.g., when we read it for scheduling: https://github.com/airbytehq/airbyte/blob/master/airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java#L50) by taking the new schema if it's available, else the old schema.

This check is only for the Create API handler, to make sure we don't accidentally pass both from the frontend. Probably would be fine to tolerate, but since nobody is passing the new schema yet, seemed better to be conservative?

Reply:
oh right, forgot this was in the context of creating connection. i think what you have is very reasonable then